pkg/ccl/sqlccl/restore.go

```diff
@@ -775,64 +638,98 @@ func Restore(
 	maxConcurrentImports := clusterNodeCount(p.ExecCfg().Gossip) * runtime.NumCPU()
 	importsSem := make(chan struct{}, maxConcurrentImports)
 
-	log.Eventf(ctx, "commencing import of data with concurrency %d", maxConcurrentImports)
+	log.Eventf(restoreCtx, "commencing import of data with concurrency %d", maxConcurrentImports)
 	tBegin := timeutil.Now()
 
-	mu := struct {
-		syncutil.Mutex
-		res               roachpb.BulkOpSummary
-		requestsCompleted []bool
-		lowWaterMark      int
-	}{
-		requestsCompleted: make([]bool, len(importRequests)),
-		lowWaterMark:      -1,
-	}
-
-	progressLogger := jobProgressLogger{
-		job:         job,
-		totalChunks: len(importRequests),
-		progressedFn: func(ctx context.Context, details interface{}) {
-			switch d := details.(type) {
-			case *jobs.Payload_Restore:
-				mu.Lock()
-				if mu.lowWaterMark >= 0 {
-					d.Restore.LowWaterMark = importRequests[mu.lowWaterMark].Key
+	// We're about to start off one goroutine that serially presplits & scatters
+	// each import span. Once split and scattered, the span is submitted to
+	// importRequestsCh to indicate it's ready for Import. Since import is so
+	// much slower, we buffer the channel to keep the split/scatter work from
+	// getting too far ahead. This both naturally rate limits the split/scatters
+	// and bounds the number of empty ranges crated if the RESTORE fails (or is
+	// cancelled).
+	const presplitLeadLimit = 10
+	importRequestsCh := make(chan *roachpb.ImportRequest, presplitLeadLimit)
```

Comment:
I feel like you could make `presplitLeadLimit = maxConcurrentImports`, under the reasoning that if every ImportRequest finishes at the exact same moment, you might as well have work for all of them to do.